src/display/api.js

```diff
@@ -1742,7 +1752,12 @@ var WorkerTransport = (function WorkerTransportClosure() {
             var size = width * height;
             var rgbaLength = size * 4;
             var buf = new Uint8Array(size * components);
-            var tmpCanvas = createScratchCanvas(width, height);
+            // make sure that this code is not executing in node.js, as
+            // it's using DOM image, and there is no library to support that.
+            assert(typeof document !== 'undefined');
```

Comment:
Looking at the entire `JpegDecode` code here, please see https://github.com/mozilla/pdf.js/blob/master/src/display/api.js#L1724-L1769, the use of `assert` seems slightly out of place here.

@yurydelendik I'd think it'd be more in line with the rest of this code if we did
```javascript
if (typeof document === 'undefined') {
  // Make sure that this code is not executing in Node.js, as
  // it's using DOM image, and there is no library to support that.
  return Promise.reject(new Error('"document" is not defined.'));
}
```
at e.g. line https://github.com/mozilla/pdf.js/blob/master/src/display/api.js#L1728 instead.